tests/library/trace-viewer.spec.ts

```diff
@@ -1439,6 +1439,32 @@ test.skip('should allow showing screenshots instead of snapshots', async ({ runA
   await expect(screenshot).toBeVisible();
 });
 
+test('canvas clipping', async ({ runAndTrace, page, server }) => {
+  const traceViewer = await runAndTrace(async () => {
+    await page.goto(server.PREFIX + '/screenshots/canvas.html#canvas-on-edge');
+    await page.waitForTimeout(1000); // ensure we could take a screenshot
```

Comment:
I predict this will be flaky 😄 We usually do a few rafs in the hopes it will redraw, but even that does not always help.

Reply:
What's a raf?

Reply:
It's [`requestAnimationFrame`](https://developer.mozilla.org/en-US/docs/Web/API/Window/requestAnimationFrame)! See https://github.com/Skn0tt/playwright/blob/ea4116e64546dca6b41160c50a42c69e86cc3fcd/tests/page/pageTest.ts#L39.